torch/fx/experimental/symbolic_shapes.py

```diff
@@ -163,6 +164,20 @@ def is_concrete_int(a: Union[int, SymInt]):
 
     return False
 
+def guard_size_oblivious(expr: Union[torch.SymBool, bool]) -> bool:
+    """
+    Perform a guard on a symbolic boolean expression in a size oblivious way.
+    This is typically used when a non-oblivious test would result in a guard
+    on a data dependent value of which we don't know the value of at compile time.
+    When a guard is tested this way, we may diverge in behavior from how regular
+    PyTorch semantics would treat it.  For more information, see
+    https://github.com/pytorch/pytorch/pull/118579
+    """
+    if isinstance(expr, torch.SymBool):
+        return expr.node.guard_size_oblivious("", 0)
+    else:
```

Comment:
should we also coerce Symints to Symbools here?

Reply:
If you're worried about people disrespecting the type signature, I can just make it error if you feed me not a bool.